tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/Filer.java

```diff
@@ -40,100 +39,80 @@ public Filer setLanguagePlugin(LanguagePlugin languagePlugin) {
-                                        .addSingleLineComment("\\/\\/")
-                                        .addMultiLineComment("\\/\\*+", "\\*+\\/")
-                                        .build()))
-                .add(
-                        new CommentStyleFileFilter(
-                                // XML comment syntax
-                                CommentSyntax.newBuilder()
-                                        .addMultiLineComment("/\\*", "\\*/")
-                                        .build()))
-                //TODO: add rest of the comment syntaxes
-                .build();
+    
+    public void visitFile(Path source, Path relativePath) {
+        logger.info("Looking into file: {} ", source);
+        String skipRegex = "\\.tmcrc|metadata\\.yml|(.*)Hidden(.*)";
+        if (source.getFileName().toString().matches(skipRegex)) {
+            logger.info("Skipping file: {} ", source);
+            return;
+        }
+        Path destination = toPath.resolve(relativePath);
+        try {
+            copyWithFilters(source, destination);
+        } catch (IOException ex) {
+            throw new IllegalStateException(ex);
+        }
     }
 
-    protected boolean skipFile(Path file) {
-        List<String> nameSkipList =
-                Arrays.asList(new String[] {".tmcrc", "metadata.yml"});
-        for (String item : nameSkipList) {
-            if (file.getFileName().toString().equals(item)) {
-                return true;
-            }
+    public void copyWithFilters(Path source, Path destination) throws IOException {
+        List<String> data = readFile(source);
+        if (!isTextFile(source)) {
+            logger.info("Just copying file from: {} to:{}", source, destination);
+        } else {
+            data = prepareFile(data, getFileExtension(source));
+            logger.info("Filtered file while copying from: {} to:{}", source, destination);
         }
-
-        nameSkipList =
-                Arrays.asList(new String[] {"Hidden"});
-        for (String item : nameSkipList) {
-            if (file.getFileName().toString().contains(item)) {
-                return true;
-            }
+        
+        if (data.isEmpty()) {
+            logger.info("skipped file as empty while copying from: {} to:{}", source, destination);
+        } else {
+            Files.createDirectories(destination.getParent());
+            Files.write(destination, data, StandardCharsets.UTF_8, StandardOpenOption.CREATE);
         }
-        return false;
+    }
+    
+    public List<String> prepareFile(List<String> data, String fileType) {
+        List<MetaSyntax> syntaxes = MetaSyntaxGenerator.listSyntaxes(fileType);
+        for (MetaSyntax metaSyntax : syntaxes) {
+            data = filterData(data, metaSyntax);
+        }
+        return data;
+    }
+    List<String> filterData(List<String> data, MetaSyntax m) {
+        return data; // Subclasses override this method
```

Comment:
The filterData method is new. Before restructing this class the prepareFile method was always overridden by subclasses. I'll try to make the whole Filer class abstract and see if it works.